core/src/sumeragi/main_loop.rs

```diff
@@ -975,6 +1015,12 @@ fn vote_for_block(
     let role = sumeragi.current_topology.role(&sumeragi.peer_id);
     trace!(%addr, %role, block_hash=%block_hash, "Block received, voting...");
 
+    let block_view_change_index = block.payload().header().view_change_index;
+    if block_view_change_index != current_view_change_index {
+        warn!(%role, %current_view_change_index, %block_view_change_index, "Could not vote for block because of incorrect view change index");
+        return None;
+    }
+
```

Comment:
Why peer can't accept block with higher view change index if it has all necessary signatures?

Reply:
It makes it easier to debug. It's a rare and unnecessary case and we save ourselves time be not supporting it.